src/formatting/formatter.rs

```diff
@@ -191,13 +199,14 @@ impl Formatter for StrFormatter {
                 let width = text.chars().count();
                 Ok(match (self.rot_interval_ms, self.init_time) {
                     (Some(rot_interval_ms), Some(init_time)) if width > self.max_width => {
-                        let width = width + 1; // Now we include '|' at the end
+                        let rot_separator = self.rot_separator.clone().unwrap_or("|".to_string());
+                        let width = width + rot_separator.len(); // Now we include `rot_separator` at the end
```

Comment:
This should be `.chars().count()` to work better with unicode (although ideally we should probably count [graphemes](https://docs.rs/unicode-segmentation/latest/unicode_segmentation/trait.UnicodeSegmentation.html#tymethod.graphemes)).